common/src/main/java/com/wynntils/utils/render/RenderUtils.java

```diff
@@ -881,6 +881,69 @@ public static void renderCustomNametag(
         }
     }
 
+    public static void renderProfessionBadge(
+            PoseStack matrixStack,
+            EntityRenderDispatcher dispatcher,
+            Entity entity,
+            ResourceLocation tex,
+            float width,
+            float height,
+            int uOffset,
+            int vOffset,
+            int u,
+            int v,
+            int textureWidth,
+            int textureHeight,
+            float customOffset,
+            float horizontalShift,
+            float verticalShift) {
+        double d = dispatcher.distanceToSqr(entity);
+        if (d <= 4096.0) {
```

Comment:
I recognize this code from MC I think, partly because of this line, partly because you named PoseStack "matrixStack". Please rename it to `poseStack`. Also, I am not sure where the render nametag event is fired, maybe we don't need to do the distance check twice.

Reply:
Vanilla Minecraft does not do a distance check. This code is from legacy.